src/ledger.ts

```diff
@@ -76,6 +82,52 @@ export class LedgerCanister {
     );
   };
 
+  /**
+   * Creates a subaccount with the name and returns the Subaccount details
+   *
+   * TODO: Does this belong here in the "LedgerCanister"
+   * TODO: Error messages
+   * TODO: Why is calling to `add_account` needed?
+   */
+  public createSubaccount = async ({
+    subAccountName,
+  }: {
+    subAccountName: string;
+  }): Promise<SubAccountDetails> => {
+    const service: NNSDappService = Actor.createActor<NNSDappService>(
+      certifiedIdlFactory,
+      {
+        agent: this.agent,
+        canisterId: this.canisterId,
+      }
+    );
+
+    const MAX_TRIES = 2;
+    const counter = 0;
+    let response: CreateSubAccountResponse = await service.create_sub_account(
+      subAccountName
+    );
+    while (response.AccountNotFound === null && counter < MAX_TRIES) {
```

Comment:
Quite risky to do it here. If we move retry logic (for example in a store) then we could avoid having some background processes w/o visual feedback after the user navigates to another page.

Reply:
that's a very good point. I would like to avoid this while as well. I added it here, to raise the question. Not sure we should merge with this while, here nor in the app. Specially not without understanding why we need it.